packages/storefront-sdk/src/client/index.ts

```diff
@@ -1,15 +1,77 @@
 import { createClient } from '@urql/core';
 import type { Client as UrqlClient } from '@urql/core';
 import type { StorefrontClientParams } from '../index.js';
+import type {
+	SetConfigParams,
+	SetConfigResponse,
+	StorefrontConfig
+} from '../types/config.js';
 
 export class StorefrontClient {
-	readonly #graphqlClient: UrqlClient;
+	#graphqlClient: UrqlClient;
+	#storefrontEndpoint: string;
+	#previewToken: string | undefined;
+	#currencyCode: string | undefined;
+	#locale: string | undefined;
+	#fetchClient: typeof globalThis.fetch;
 
 	constructor(params: StorefrontClientParams) {
+		this.#storefrontEndpoint = params.storefrontEndpoint;
+		this.#previewToken = params.previewToken;
+		this.#fetchClient = params.fetchClient ?? globalThis.fetch;
+		this.#currencyCode = params.currencyCode;
+		this.#locale = params.locale;
+
+		let headers = {};
+		if (this.#previewToken) {
+			headers = { X_NACELLE_PREVIEW_TOKEN: this.#previewToken };
```

Comment:
This header value is expected to use dashes:

```suggestion
			headers = { x-nacelle-space-token: this.#previewToken };
```